packages/swap/src/routes/v2/swap.ts

```diff
@@ -0,0 +1,323 @@
+import express from 'express';
+import { assetConstants, getAssetAndChain } from '@/shared/enums';
+import { getPriceFromPriceX128 } from '@/shared/functions';
+import { getRequiredBlockConfirmations } from '@/swap/utils/rpc';
+import {
+  depositChannelInclude,
+  getDepositIgnoredFailedState,
+  getEgressStatusFields,
+  getSwapFields,
+  getSwapState,
+  swapRequestInclude,
+} from './utils';
+import prisma, { Prisma, SwapFee } from '../../client';
+import { getPendingDeposit } from '../../ingress-egress-tracking';
+import { readField } from '../../utils/function';
+import logger from '../../utils/logger';
+import ServiceError from '../../utils/ServiceError';
+import {
+  channelIdRegex,
+  coerceChain,
+  swapRequestId,
+  txHashRegex,
+  estimateSwapDuration,
+  isEgressableSwap,
+} from '../../utils/swap';
+import { asyncHandler } from '../common';
+
+const router = express.Router();
+
+export enum StateV2 {
+  Failed = 'FAILED',
+  Complete = 'COMPLETE',
```

Comment:
just a nitpick, maybe past tense would be better here as well.
```suggestion
  Completed = 'COMPLETED',
```